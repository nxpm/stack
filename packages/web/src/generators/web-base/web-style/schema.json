{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "WebStyle",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "appName": {
      "type": "string",
      "description": "Name of the app to connect to",
      "alias": "a"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "library": {
      "description": "The UI library",
      "type": "string",
      "default": "tailwind",
      "x-prompt": {
        "message": "Which type of library would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "bootstrap",
            "label": "Bootstrap"
          },
          {
            "value": "tailwind",
            "label": "Tailwind"
          }
        ]
      }
    }
  },
  "required": ["appName", "name"]
}
