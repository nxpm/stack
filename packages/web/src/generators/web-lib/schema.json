{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "id": "ApiLib",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "type": {
      "type": "string",
      "enum": ["data-access", "feature", "util"],
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": "What type would you like to create?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": ["name"]
}
