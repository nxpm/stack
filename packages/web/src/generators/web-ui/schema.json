{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "id": "WebUi",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "library": {
      "description": "The UI library",
      "type": "string",
      "default": "tailwind",
      "x-prompt": {
        "message": "Which type of library would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "tailwind",
            "label": "Tailwind"
          }
        ]
      }
    }
  },
  "required": ["name"]
}
