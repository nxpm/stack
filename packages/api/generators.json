{
  "$schema": "http://json-schema.org/schema",
  "name": "api",
  "version": "0.0.1",
  "generators": {
    "api": {
      "factory": "./src/generators/api/generator-api#generatorApi",
      "schema": "./src/generators/api/schema.json",
      "description": "api generator"
    },
    "api-e2e": {
      "factory": "./src/generators/api-e2e/generator-api-e2e#generatorApiE2e",
      "schema": "./src/generators/api-e2e/schema.json",
      "description": "api-e2e generator"
    },
    "api-feature": {
      "factory": "./src/generators/api-feature/generator-api-feature#generatorApiFeature",
      "schema": "./src/generators/api-feature/schema.json",
      "description": "api-feature generator"
    },
    "api-lib": {
      "factory": "./src/generators/api-lib/generator-api-lib#generatorApiLib",
      "schema": "./src/generators/api-lib/schema.json",
      "description": "api-lib generator"
    }
  }
}
