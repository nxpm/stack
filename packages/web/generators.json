{
  "$schema": "http://json-schema.org/schema",
  "name": "web",
  "version": "0.0.1",
  "generators": {
    "web": {
      "factory": "./src/generators/web/generator-web#generatorWeb",
      "schema": "./src/generators/web/schema.json",
      "description": "web generator"
    },
    "web-base": {
      "factory": "./src/generators/web-base/generator-web-base#generatorWebBase",
      "schema": "./src/generators/web-base/schema.json",
      "description": "web-base generator"
    },
    "web-feature": {
      "factory": "./src/generators/web-feature/generator-web-feature#generatorWebFeature",
      "schema": "./src/generators/web-feature/schema.json",
      "description": "web-feature generator"
    },
    "web-lib": {
      "factory": "./src/generators/web-lib/generator-web-lib#generatorWebLib",
      "schema": "./src/generators/web-lib/schema.json",
      "description": "web-lib generator"
    },
    "web-ui": {
      "factory": "./src/generators/web-ui/generator-web-ui#generatorWebUi",
      "schema": "./src/generators/web-ui/schema.json",
      "description": "web-ui generator"
    },
    "web-ui-lib": {
      "factory": "./src/generators/web-ui-lib/generator-web-ui-lib#generatorWebUiLib",
      "schema": "./src/generators/web-ui-lib/schema.json",
      "description": "web-ui-lib generator"
    }
  }
}
