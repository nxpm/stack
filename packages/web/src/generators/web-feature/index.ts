export * from './generate-web-feature-about'
export * from './generate-web-feature-account'
export * from './generate-web-feature-admin'
export * from './generate-web-feature-auth'
export * from './generate-web-feature-core'
export * from './generate-web-feature-dashboard'
export * from './generate-web-feature-layout'
export * from './generate-web-feature-shell'
