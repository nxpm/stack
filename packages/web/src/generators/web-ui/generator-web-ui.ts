import { Tree } from '@nrwl/devkit'
import { WebUiGeneratorSchema } from './schema'
import { generateWebUiLibButton } from './generate-web-ui-lib-button'
import { generateWebUiLibForm } from './generate-web-ui-lib-form'
import { generateWebUiLibLoader } from './generate-web-ui-lib-loader'
import { generateWebUiLibIcon } from './generate-web-ui-lib-icon'
import { generateWebUiLibPage } from './generate-web-ui-lib-page'
import { generateWebUiLibPageHeader } from './generate-web-ui-lib-page-header'
import { generateWebUiLibSidebarPage } from './generate-web-ui-lib-sidebar-page'
import { generateWebUiLibTable } from './generate-web-ui-lib-table'
import { generateWebUiLibToast } from './generate-web-ui-lib-toast'

export async function generatorWebUi(host: Tree, options: WebUiGeneratorSchema) {
  const library = 'tailwind'
  const directory = options.directory || options.name

  options = { ...options, directory, library }

  await generateWebUiLibButton(host, options)
  await generateWebUiLibForm(host, options)
  await generateWebUiLibLoader(host, options)
  await generateWebUiLibIcon(host, options)
  await generateWebUiLibPage(host, options)
  await generateWebUiLibPageHeader(host, options)
  await generateWebUiLibSidebarPage(host, options)
  await generateWebUiLibTable(host, options)
  await generateWebUiLibToast(host, options)
}
