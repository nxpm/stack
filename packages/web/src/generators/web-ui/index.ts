export * from './generate-web-ui-lib-button'
export * from './generate-web-ui-lib-form'
export * from './generate-web-ui-lib-icon'
export * from './generate-web-ui-lib-loader'
export * from './generate-web-ui-lib-page'
export * from './generate-web-ui-lib-page-header'
export * from './generate-web-ui-lib-sidebar-page'
export * from './generate-web-ui-lib-table'
export * from './generate-web-ui-lib-toast'
