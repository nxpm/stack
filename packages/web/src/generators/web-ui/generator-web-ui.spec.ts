import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing'
import { readJson, Tree } from '@nrwl/devkit'

import { generatorWebUi } from './generator-web-ui'
import { WebUiGeneratorSchema } from './schema'
import { applicationGenerator } from '@nrwl/angular/generators'

describe('web-ui generator', () => {
  let appTree: Tree
  const options: WebUiGeneratorSchema = { name: 'test', library: 'tailwind', webName: 'test' }

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should run successfully', async () => {
    await applicationGenerator(appTree, { name: options.webName })
    await generatorWebUi(appTree, options)

    const nxJson = readJson(appTree, 'nx.json')
    expect(nxJson).toMatchSnapshot()

    // const configApp = readProjectConfiguration(appTree, `${options.webName}`)
    // expect(configApp).toMatchSnapshot()
  })
})
