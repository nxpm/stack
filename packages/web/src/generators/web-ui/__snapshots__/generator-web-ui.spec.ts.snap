// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web-ui generator should run successfully 1`] = `
Object {
  "affected": Object {
    "defaultBase": "master",
  },
  "npmScope": "proj",
  "projects": Object {
    "test": Object {
      "tags": Array [],
    },
    "test-e2e": Object {
      "implicitDependencies": Array [
        "test",
      ],
      "tags": Array [],
    },
    "test-ui-button": Object {
      "tags": Array [],
    },
    "test-ui-form": Object {
      "tags": Array [],
    },
    "test-ui-icon": Object {
      "tags": Array [],
    },
    "test-ui-loader": Object {
      "tags": Array [],
    },
    "test-ui-page": Object {
      "tags": Array [],
    },
    "test-ui-page-header": Object {
      "tags": Array [],
    },
    "test-ui-sidebar-page": Object {
      "tags": Array [],
    },
    "test-ui-table": Object {
      "tags": Array [],
    },
    "test-ui-toast": Object {
      "tags": Array [],
    },
  },
  "tasksRunnerOptions": Object {
    "default": Object {
      "options": Object {
        "cacheableOperations": Array [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "@nrwl/workspace/tasks-runners/default",
    },
  },
}
`;
